tezos/api/src/ocaml_conv/to_ocaml.rs

```diff
@@ -0,0 +1,297 @@
+        chain_arg: OCamlBytes,
+        request: JsonRpcRequest,
+        ffi_service: FfiRpcService,
+    ) -> ProtocolJsonRpcRequest;
+
+    // Headers
+
+    alloc fn alloc_block_header_shell_header(
+        level: OCamlInt32,
+        proto_level: OCamlInt,
+        predecessor: OCamlBlockHash,
+        timestamp: OCamlInt64,
+        validation_passes: OCamlInt,
+        operations_hash: OCamlOperationListListHash,
+        fitness: OCamlList<OCamlBytes>,
+        context: OCamlContextHash,
+    ) -> BlockHeaderShellHeader;
+
+    alloc fn alloc_block_header(
+        shell: BlockHeaderShellHeader,
+        protocol_data: OCamlBytes,
+    ) -> BlockHeader;
+
+    alloc fn alloc_operation_shell_header(
+        branch: OCamlBlockHash,
+    ) -> OperationShellHeader;
+
+    // Operation
+
+    alloc fn alloc_operation(
+        shell: OperationShellHeader,
+        proto: OCamlBytes,
+    ) -> Operation;
+
+    // Wrappers
+
+    alloc fn alloc_prevalidator_wrapper(
+        chain_id: OCamlBytes,
+        protocol: OCamlProtocolHash,
+    ) -> PrevalidatorWrapper;
+
+    // Hashes
+
+    alloc fn alloc_operation_list_list_hash(hash: OCamlBytes) -> OCamlOperationListListHash;
+    alloc fn alloc_operation_hash(hash: OCamlBytes) -> OCamlOperationHash;
+    alloc fn alloc_block_hash(hash: OCamlBytes) -> OCamlBlockHash;
+    alloc fn alloc_context_hash(hash: OCamlBytes) -> OCamlContextHash;
+    alloc fn alloc_protocol_hash(hash: OCamlBytes) -> OCamlProtocolHash;
+}
+
+unsafe impl ToOCaml<ApplyBlockRequest> for ApplyBlockRequest {
+    fn to_ocaml(&self, token: OCamlAllocToken) -> OCamlAllocResult<ApplyBlockRequest> {
+        let operations: Vec<Vec<FfiOperation>> = self
+            .operations
+            .iter()
+            .map(|ops| ops.iter().map(|op| FfiOperation(op.clone())).collect())
+            .collect();
+        unsafe {
+            alloc_apply_block_request(
+                token,
+                &self.chain_id,
+                &FfiBlockHeader(self.block_header.clone()),
+                &FfiBlockHeader(self.pred_header.clone()),
+                &self.max_operations_ttl,
+                &operations,
+            )
+        }
+    }
+}
+
+unsafe impl ToOCaml<ApplyBlockResponse> for ApplyBlockResponse {
+    fn to_ocaml(&self, token: OCamlAllocToken) -> OCamlAllocResult<ApplyBlockResponse> {
+        unsafe {
+            alloc_apply_block_response(
+                token,
+                &self.validation_result_message,
+                &self.context_hash,
+                &self.block_header_proto_json,
+                &self.block_header_proto_metadata_json,
+                &self.operations_proto_metadata_json,
+                &self.max_operations_ttl,
+                &self.last_allowed_fork_level,
+                &self.forking_testchain,
+                &self.forking_testchain_data,
+            )
+        }
+    }
+}
+
+unsafe impl ToOCaml<ForkingTestchainData> for ForkingTestchainData {
```

Comment:
@tizoc 
if there is inside-block wrapped with unsafe, do we need also `unsafe impl` ?